pkg/controller/endpointslice/endpointslice_controller_test.go

```diff
@@ -101,6 +101,25 @@ func TestSyncServiceNoSelector(t *testing.T) {
 	assert.Len(t, client.Actions(), 0)
 }
 
+// Ensure SyncService for service with pending deletion results in no action
```

Comment:
is it worth separately testing that update/delete still occurs?

Reply:
I've added test coverage for this in `reconciler_test.go`. It's actually a relatively complex thing to test since the combination of creates and deletes will be converted into updates, so testing there was slightly more straightforward.